### DISCUSSION TITLE
API key shown in Network Tab

### DISCUSSION THREAD

**Discussion by jayehernandez** [supabase/supabase#256] [Questions] [answered]
Hi, just want to ask if the `apikey` is intentionally shown in the Request Headers of the Network Tab in devtools whenever I call from the API? Is this the default behavior, or are there any best practices for this that I might have missed?

For reference too, I'm using `supabase-js`. 

<img width="474" alt="Screen Shot 2020-10-19 at 7 39 09 PM" src="https://user-images.githubusercontent.com/13959651/96445750-fb687300-1242-11eb-8123-3d7b75b0771a.png">

**Comment by kiwicopple** [comment, marked as answer]
Hi @jayehernandez - yes this is fine! Just make sure you're only using the `anon` key, and not the `service` key.

The `anon` key adheres to Row Level Security, so you can add Policies to each table.

The `service` key will completely bypass Row Level Security. Make sure you never expose this on a browser/client.

Soon we will give you the ability to add new keys too.

**Reply by jayehernandez** [reply]
A bit late on the reply, but thank you for clearing this up!

**Reply by Ranzeplay** [reply]
Is it possible to add new keys now?

**Comment by w0wka91** [comment]
I don't really get the point of the API key when it is publicly available and can be used by everyone. Wouldn't it be enough to just provide the bearer token? Did I miss something?

**Reply by jdgamble555** [reply]
The API key is public.  Anything in the client is public.  From my understanding SupabaseJS needs the API key (anon), while the bearer token would be more for REST or GraphQL endpoint.

Here is details on the same question for a Firebase API key:

https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public

J
